### ISSUE TITLE
Notify loudly when ECS is still scaled up

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#720]
In GitLab by @infinitewarp on Jul 22, 2020, 15:15

## Summary

As a cloudigrade sysop, I expect our ECS cluster to only be scaled up for a few minutes at a time, generally just long enough to define, attach, and run the task for houndigrade. If houndigrade fails to enqueue any results or our results processing fails unexpectedly, we may unintentionally leave the cluster scaled up forever.

We current log at `info` level when we attempt to scale up but find that the cluster is already scaled up, and then our code simply returns, only to be restarted at the next scheduled run. We should consider raising this to `error` *if* we believe a reasonably long amount of time has elapsed. We should not *always* raise to `error`, though, because it's possible for a normal healthy process to have tasks align in such a way that operations related to scaling up and down run too soon, such as when we are *actually* waiting for the cluster to scale up and run houndigrade.

## Acceptance Criteria

- [ ] Verify that a message is logged at error and alerts Sentry and Slack #doppler_alerts when `scale_up_inspection_cluster` runs but the cluster appears to have been up for too long.
  - "Too long" is an implementation detail that we need to document here before verifying.
  - "Too long" could be a length of time or it could be based on checking for completed tasks.
  - [ ] If we use a simple time check for this, make that amount of time configurable with the usual environment variable and OpenShift e2e-deploy config map stuff.
- [ ] Verify this manually at least once, but *not* with QE's automated integration tests.

## Assumptions and Questions

- To be clear: We are only alerting. We are not **yet** automatically forcing the cluster to scale down. We may implement a forced scale down in a future issue. We need to gather information first.
- We discussed the possibility of interrogating our "retry" wrappers to know whether or not this operation has tried and failed (thus retried) and only alert if it's a retry. However, this `scale_up_inspection_cluster` task *is not currently* decorated with `retriable_shared_task`. This lack of retry is **correct** by design. **Scheduled tasks should not retry.**
- Possible implementation detail: When we ask AWS if the cluster is scaled up, we should be able to identify the EC2 instance being used by the ECS cluster. Check how long the instance has been up, and use that to determine if we should alert.
- Possible implementation detail: When we ask AWS if the cluster is scaled up, we should be able to identify the state of any task for houndigrade we gave the cluster. Hypothetically, we could see if the task we gave it completed running. If we can reliably see completed tasks, we can use this to determine if we should alert.
  - However, @infinitewarp has seen clusters completely forget about past completed tasks. We are unsure how long that might take. We should experiment and try to verify this. If we can identify completed tasks reliably, consider leveraging this idea!

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Aug 18, 2020, 13:50

I've experimented with various ECS commands to try to reliably identify if a task has run *after* it has run, but unfortunately my earlier comment appears to be true. Once an instance of a task definition has completed execution on a cluster, its record is cleaned up, making it difficult to know what stage we are in the process.

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Aug 19, 2020, 15:18

Sentry issue: [CLOUDIGRADE-CELERY-7P](https://sentry.io/organizations/cloudigrade/issues/1849022109/?referrer=gitlab_integration)

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Aug 19, 2020, 15:19

That aforementioned Sentry issue is a side-effect of the refactoring done in !782.

**Comment by katherine-black** [comment]
In GitLab by @pakamble on Aug 28, 2020, 05:00

As a QE I have manually verified this story by running runnning following test steps.

1. Created a 10 instances on AWS with using 10 different images.
2. Added sources for cloudmeter.
3. wait until all image inspection is getting complete.

**Observation.**

1. While doing inspection process ECS cluster is scaled up beyond permissible time limit hence cloudmeter logged error message. (refer following log snippet from kibana.)

![kibana](/uploads/abae7afa99e1808f8f1d53d166ae83ca/kibana.png) 

2. sentry has raised alert message.

![sentry](/uploads/4fcd9d04f0c3265ef37f08987c9b61ef/sentry.png)
